contracts/zxp/game/seasons/mechanics/SecretRewards.sol

```diff
@@ -33,6 +35,7 @@ contract SecretRewards is ObjectRegistryClient, ISecretRewards {
         string reveal;
     }
 
+    // what is the reason there's no nonce counter on the contract?

```

Comment:
No, could store it, just no good reason to on-contract since it is not getting used anywhere. 
This does mean you have to store these offchain, but even if you had them on-chain you'd still have some work to do there managing these.
And I don't think we'd want to spend gas on making them like enumerable or something.